app/classifier/tasks/text/editor.cjsx

```diff
@@ -0,0 +1,85 @@
+React = require 'react'
+alert = require '../../../lib/alert'
+AutoSave = require '../../../components/auto-save'
+handleInputChange = require '../../../lib/handle-input-change'
+{MarkdownEditor} = require 'markdownz'
+MarkdownHelp = require '../../../partials/markdown-help'
+NextTaskSelector = require '../next-task-selector'
+
+module?.exports = React.createClass
+  displayName: 'TextTaskEditor'
+
+  getDefaultProps: ->
+    workflow: {}
+    task: {}
+
+  tagExists:(tag) ->
+    if @props.task.text_tags
+      @props.task.text_tags.indexOf(tag) > -1
+
+  updateTags:(e)->
```

Comment:
not sure if properly captures possibilities or if really any improvement, maybe there's an even better option, but i think a different way to write could be:
```
value = e.target.value
checked = e.target.checked
text_tags = @props.task.text_tags ? []
if text_tags.indexOf(value) is -1 and checked
  text_tags.push(value)
else if !checked
  removalIndex = text_tags.indexOf(value)
  text_tags.splice(removalIndex,1);
@props.task.text_tags = text_tags
@props.onChange @props.task